frontend/app/src/types/matching.ts

```diff
@@ -0,0 +1,19 @@
+import { z } from "zod";
+
+import { MATCHING_EVENTS } from "@/constants/matching";
+import { DifficultyType, TopicTagType } from "@/constants/question";
+
+export type MatchingStatusType =
+  (typeof MATCHING_EVENTS)[keyof typeof MATCHING_EVENTS];
+
+export type Preferences = {
+  difficulty: DifficultyType[];
+  category: TopicTagType[];
+};
+
+export const matchingSchema = z.object({
+  status: z.nativeEnum(MATCHING_EVENTS),
```

Comment:
nice! didn't know that passing `as const` objects works as well

https://github.com/colinhacks/zod#native-enums